trinity/utils/shutdown.py

```diff
@@ -12,16 +18,23 @@
 
 async def exit_on_signal(service_to_exit: BaseService, endpoint: Endpoint = None) -> None:
     loop = service_to_exit.get_event_loop()
+    async with wait_for_exit(loop):
+        await service_to_exit.cancel()
+        if endpoint is not None:
+            endpoint.stop()
+        service_to_exit._executor.shutdown(wait=True)
+
+
+@asynccontextmanager
+async def wait_for_exit(loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
```

Comment:
I think this default value is going to be problematic. It will be evaluated once at import time, which is probably not be desired. So far all method calls explicitly set the loop, anyway.

How about we require the loop to be passed for now. If leaving it out becomes convenient at some point in the future, we can re-add it. (The default lookup would go in the method body, to detect the loop at runtime).

Reply:
> It will be evaluated once at import time

Oh, I assumed that would only be evaluated in the case it is called without that parameter. Didn't know these things are evaluated eagerly in Python (Still a Python noob after all).

>How about we require the loop to be passed for now

Yes, let's do that.